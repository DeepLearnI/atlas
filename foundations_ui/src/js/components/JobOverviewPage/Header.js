import React from 'react';
import PropTypes from 'prop-types';
import BaseActions from '../../actions/BaseActions';
import CommonActions from '../../actions/CommonActions';

class Header extends React.Component {
  constructor(props) {
    super(props);
    let name = '';
    let createdAt = '';
    let owners = '';

    if (this.props.location.state && this.props.location.state.project && this.props.location.state.project !== {}) {
      name = this.props.location.state.project.name;
      createdAt = this.props.location.state.project.created_at;
      owners = this.props.location.state.project.owner;
    }

    this.state = {
      name: name,
      dateCreated: createdAt,
      projectOwners: owners,
    };

    this.onClickBack = this.onClickBack.bind(this);
    this.onKeyDown = this.onKeyDown.bind(this);
  }

  async reload() {
    const { name, dateCreated, projectOwners } = this.state;

    if (name === '' || dateCreated === '' || projectOwners === '') {
      const { projectName } = this.props.match.params;
      const fetchedProjects = await BaseActions.getFromStaging('projects');
      const selectedProject = fetchedProjects.filter(item => item.name === projectName)[0];

      if (selectedProject && selectedProject.length > 0) {
        this.setState({
          name: selectedProject.name,
          dateCreated: selectedProject.created_at,
          projectOwners: selectedProject.owner,
        });
      }
    }
  }

  componentDidMount() {
    this.reload();
  }

  onClickBack() {
    const { history } = this.props;
    history.push(
      '/projects',
    );
  }

  onKeyDown() {}

  render() {
    const { name, dateCreated, projectOwners } = this.state;

    return (
      <div>
        <div className="job-overview-header-container">
          <div>
            <h3
              className="text-project-directory"
              onClick={this.onClickBack}
              aria-label="Go Back"
              onKeyDown={this.onKeyDown}
            >
              <i
                className="icono-arrow1-right"
              />
              Project Directory
            </h3>
            <h1 className="font-bold">{name}</h1>
          </div>
          <div>
            <div className="container-label-date font-bold">Date Created: </div>
            <div className="container-text-date">{CommonActions.formatDate(dateCreated)}</div>
          </div>
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,

};

Header.defaultProps = {
  history: {},
  location: {},
  match: { params: {} },
};

export default Header;
