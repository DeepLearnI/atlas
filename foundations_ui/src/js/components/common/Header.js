import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProjectHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      numProjects: this.props.numProjects,
      pageTitle: this.props.pageTitle,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ numProjects: nextProps.numProjects });
  }

  render() {
    const { numProjects, pageTitle } = this.state;

    let projectCount;

    if (pageTitle === 'Projects') {
      projectCount = (
        <div className="half-width inline-block text-right">
          <h2 className="blue-border-bottom font-bold">
              Total Projects: <span>{numProjects}</span>
          </h2>
        </div>
      );
    }


    return (
      <div className="project-header-container">
        <div className="project-header-info-container">
          <div className="project-header-total-projects-container">
            <div className="half-width inline-block">
              <h1 className="blue-border-bottom font-bold">{pageTitle}</h1>
            </div>
            {projectCount}
            <div className="project-header-sort-filter-container" />
          </div>
        </div>
      </div>
    );
  }
}

ProjectHeader.propTypes = {
  numProjects: PropTypes.number,
  pageTitle: PropTypes.string,
};

ProjectHeader.defaultProps = {
  numProjects: 0,
  pageTitle: 'Projects',
};

export default ProjectHeader;
