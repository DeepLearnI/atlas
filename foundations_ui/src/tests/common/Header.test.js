import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../../js/components/common/Header';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders ProjectHeader', () => {
  const wrapper = shallow(<Header/>);
  expect(wrapper).toMatchSnapshot();
});

it('Has Projects', () => {
  const numProjects = 3;
  const wrapper = mount(<Header pageTitle='Projects' numProjects={numProjects}/>); 
  const state = wrapper.state();
  expect(state.numProjects > 0);
});

it('Has No Projects', () => {
    const wrapper = mount(<Header pageTitle='Login'/>); 
    const state = wrapper.state();
    expect(state.numProjects === 0);
});

it('Has default pageTitle of Projects', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.state.pageTitle === 'Projects')
})