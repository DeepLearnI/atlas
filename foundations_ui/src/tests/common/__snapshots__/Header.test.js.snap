// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders ProjectHeader 1`] = `
<div
  className="project-header-container"
>
  <div
    className="project-header-logo-container"
  >
    <div
      className="i--icon-logo"
    />
    <h2
      className="font-bold"
    >
      Foundations
    </h2>
  </div>
  <div
    className="project-header-info-container"
  >
    <div
      className="project-header-total-projects-container"
    >
      <div
        className="half-width inline-block"
      >
        <h1
          className="blue-border-bottom font-bold"
        >
          Projects
        </h1>
      </div>
      <div
        className="half-width inline-block text-right"
      >
        <h2
          className="blue-border-bottom font-bold"
        >
          Total Projects: 
          <span>
            0
          </span>
        </h2>
      </div>
      <div
        className="project-header-sort-filter-container"
      />
    </div>
  </div>
</div>
`;
